bpo-45442: Add deactivate step to venv tutorial.

ShivnarenSrinivasan opened the pull request:
Document process to deactivate virtual environment.

<!-- issue-number: [bpo-45442](https://bugs.python.org/issue45442) -->
https://bugs.python.org/issue45442
<!-- /issue-number -->

ShivnarenSrinivasan committed:
bpo-45442: Add deactivate step to venv tutorial.

vstinner committed:
bpo-45474: Fix the limited C API of marshal.h (GH-28956)

Remove two functions from the limited C API:

* PyMarshal_WriteLongToFile()
* PyMarshal_WriteObjectToFile()

The PEP 384 excludes functions expecting "FILE*" from the stable ABI.

Remove also the Py_MARSHAL_VERSION macro from the limited C API.

vstinner committed:
bpo-45434: Limited Python.h no longer includes stdio.h (GH-28960)

The <Python.h> header file no longer includes <stdio.h> if the
Py_LIMITED_API macro is defined.

vstinner committed:
bpo-41710: Fix What's New Entry credit (GH-28962)

Fix bad copy/paste.

vstinner committed:
bpo-45434: Remove useless space in includes (GH-28963)

Micro-optimize spaces!

vstinner committed:
po-35134: Move Include/funcobject.h to Include/cpython/ (GH-28958)

Remove redundant "#ifndef Py_LIMITED_API" in funcobject.h.

vstinner committed:
bpo-35134: Move Include/cellobject.h to Include/cpython/ (GH-28964)

benjaminp committed:
closes bpo-45479: Degunkify Py_UniversalNewlineFgets. (GH-28965)

Remove dead variables and control flow.

benjaminp committed:
bpo-45479: Futher simplify Py_UniversalNewlineFgets. (GH-28967)

Thank you to Eryk Sun for the suggestions in https://github.com/python/cpython/pull/28965#discussion_r729527143.

tirkarthi commented on Doc/tutorial/venv.rst:
```diff
@@ -88,6 +88,9 @@ For example:
   '~/envs/tutorial-env/lib/python3.5/site-packages']
   >>>
 
+To deactivate a virtual environment, type::
```
Error in CI : https://dev.azure.com/Python/cpython/_build/results?buildId=89615&view=logs&j=4db1505a-29e5-5cc0-240b-53a8a2681f75&t=a975920c-8356-5388-147c-613d5fab0171&l=517

```
Warning, treated as error:
/home/vsts/work/1/s/Doc/tutorial/venv.rst:93:Definition list ends without a blank line; unexpected unindent.
make: *** [Makefile:51: build] Error 2
```

ShivnarenSrinivasan replied:
Hopefully fixed with the second commit

vstinner committed:
bpo-35134: Move classobject.h to Include/cpython/ (GH-28968)

Move classobject.h, context.h, genobject.h and longintrepr.h header
files from Include/ to Include/cpython/.

Remove redundant "#ifndef Py_LIMITED_API" in context.h.

Remove explicit #include "longintrepr.h" in C files. It's not needed,
Python.h already includes it.

pradyunsg commented on Doc/tutorial/venv.rst:
```diff
@@ -88,6 +88,9 @@ For example:
   >>>
 
+To deactivate a virtual environment, type::
+  deactivate
```
```suggestion
To deactivate a virtual environment, type::

    deactivate
```

Please see https://devguide.python.org/documenting/#source-code and https://pradyunsg.me/furo/reference/code-blocks/ for how code blocks in Sphinx work.

ShivnarenSrinivasan replied:
Thanks for the tip

ginggs committed:
bpo-45428: Fix reading filenames from stdin in py_compile (GH-28848)

Strip trailing '\n'.

ShivnarenSrinivasan committed:
bpo-45442: Fix indentation/syntax error

Co-authored-by: Pradyun Gedam <pradyunsg@gmail.com>

vstinner committed:
bpo-35081: Move interpreteridobject.h to Include/internal/ (GH-28969)

Move the interpreteridobject.h header file from Include/ to
Include/internal/. It only provides private functions.

vsajip commented on Doc/tutorial/venv.rst:
```diff
@@ -88,6 +88,10 @@ For example:
+To deactivate a virtual environment, type::
+
+    deactivate
+into the terminal. This is a shell function defined by the script when the virtual environment is activated.
```
The specific detail about `deactivate` being a shell function isn't in general correct, though it is true on POSIX and with PowerShell. There is a corresponding `deactivate.bat` which comes into play on Windows where `cmd.exe` rather than PowerShell is used. You could just remove this detail, which isn't especially important to know from a user perspective.

ShivnarenSrinivasan replied:
Got you—will remove the line.

vstinner committed:
bpo-45434: Remove Include/eval.h header file (GH-28973)

Move Include/eval.h content into Include/ceval.h and
Include/cpython/ceval.h, and remove Include/eval.h.

pablogsal committed:
bpo-45445: Remove incorrectly commited test file (GH-28972)

vstinner committed:
bpo-44113: Move the What's New entry to Deprecate section (GH-28974)

vstinner committed:
bpo-45482: Rename namespaceobject.h to pycore_namespace.h (GH-28975)

Rename Include/namespaceobject.h to
Include/internal/pycore_namespace.h.

The _testmultiphase extension is now built with the
Py_BUILD_CORE_MODULE macro defined to access _PyNamespace_Type.

object.c: remove unused "pycore_context.h" include.

ShivnarenSrinivasan committed:
bpo-45442: Remove explanation of implementation detail.
